### PR TITLE
switch to node_modules/.cache directory instead of $HOME for binaries DL

### PR DESCRIPTION
**Pull request by guillaumervls**
Fixes https://github.com/nodkz/mongodb-memory-server/issues/95

### CODE REVIEW

**Comment by AJRdev** [comment]
@guillaumervls @nodkz I think there is a problem with the current Travis CI workflow and we have not seen this problem before because until now the cache folder of the mongodb binary was in the **%HOME/.mongodb-binaries** and h'asn't changed apparently in a while.

The Travis CI is failing because it runs the yarn test script before running the yarn build script that transpiles  the files with news changes and runs the post_install script that will download the binary in the new location (node_modules/.cache/mongodb-binaries) and cache it for yarn test script.
Since the binary right now is not cache for the test script, the binary is downloaded during the tests that causes this timeout : 
```bash
Timeout - Async callback was not invoked within the 5000ms timeout specified by jest.setTimeout.
```

Should we have launch the **yarn run build** before the **yarn run test** ?

**Comment by AJRdev** [comment]
@guillaumervls  Oh my bad, I know why it is not working, the new package find-cache-dir wasn't commited into yarn.lock file, so the Travis CI didn't launch the yarn command that would have triggered the post_install script and downloaded the mongodb binaries :)

We just need to do on the branch of the PR, yarn and then commit the changes in yarn.lock to fix the Travis pipeline.

@nodkz And yes, I agree we should fallback on the old %HOME folder until we bump a new major version.

File: src/util/MongoBinary.js [hunk 1]
```diff
@@ -24,7 +25,8 @@ export default class MongoBinary {
   static async getPath(opts?: MongoBinaryOpts = {}): Promise<string> {
     const defaultOptions = {
       downloadDir:
-        process.env?.MONGOMS_DOWNLOAD_DIR || path.resolve(os.homedir(), '.mongodb-binaries'),
```
Comment (AJRdev) [inline src/util/MongoBinary.js]: Like @nodkz suggested we should keep the fallback on **path.resolve(os.homedir(), '.mongodb-binaries')** until a new major version is bumped.

**Comment by guillaumervls** [comment]
@AJRdev @nodkz Done for the fallback. As to yarn.lock, I don't use yarn, could you tell me the update needed ?

File: src/util/MongoBinary.js [hunk 2]
```diff
@@ -23,10 +24,13 @@ export default class MongoBinary {
   static cache: MongoBinaryCache = {};
 
   static async getPath(opts?: MongoBinaryOpts = {}): Promise<string> {
+    const legacyDLDir = path.resolve(os.homedir(), '.mongodb-binaries');
     const defaultOptions = {
       downloadDir:
         process.env?.MONGOMS_DOWNLOAD_DIR ||
-        path.resolve(findCacheDir({ name: 'mongodb-memory-server' }), 'mongodb-binaries'),
+        (fs.existsSync(legacyDLDir)
```
Comment (AJRdev) [inline src/util/MongoBinary.js]: Why did you add this check ?
And not just piped everything like this : 
```javascript
downloadDir:
        process.env?.MONGOMS_DOWNLOAD_DIR ||
        path.resolve(os.homedir(), '.mongodb-binaries') ||
        path.resolve(findCacheDir({ name: 'mongodb-memory-server' }), 'mongodb-binaries')
```

**Comment by AJRdev** [comment]
@guillaumervls For yarn, you just need to install it globally with **npm install -g yarn**, and then run at the root of the project the command **yarn** (the equivalent of npm install), it will install all the dependecies and add the new one in the yarn.lock file, then you just have to commit and push it on your branch.

If you have any warning messages during the yarn command concerning the package-lock.json, just remove then package-lock.json file and re-do the procedure above.

File: src/util/MongoBinary.js (hunk 2 above)
Comment (guillaumervls) [reply src/util/MongoBinary.js]: `path.resolve(os.homedir(), '.mongodb-binaries')`
is always truthy (it just returns a path), so

`path.resolve(findCacheDir({ name: 'mongodb-memory-server' }), 'mongodb-binaries')`
would never be used

**Comment by guillaumervls** [comment]
Done

**Comment by guillaumervls** [comment]
> @guillaumervls For yarn, you just need to install it globally with **npm install -g yarn**, and then run at the root of the project the command **yarn** (the equivalent of npm install), it will install all the dependecies and add the new one in the yarn.lock file, then you just have to commit and push it on your branch.
> 
> If you have any warning messages during the yarn command concerning the package-lock.json, just remove then package-lock.json file and re-do the procedure above.
> 
> This procedure should fix the Travis CI pipeline.

**Comment by guillaumervls** [comment]
(Sorry, misclicked on close pull request)

**Comment by nodkz** [comment]
@guillaumervls @AJRdev thanks a lot for PR and fantastic work! 🍻👍

All looks good! Merged!
